src/lib.rs

```diff
@@ -25,17 +25,24 @@ mod trait_impls;
 #[repr(transparent)]
 pub struct NonEmptyString(String);
 
-#[allow(clippy::len_without_is_empty)] // is_empty would always returns false so it seems a bit silly to have it.
+#[allow(clippy::len_without_is_empty, reason = "is_empty would always returns false so it seems a bit silly to have it.")]
 impl NonEmptyString {
-    /// Attempts to create a new NonEmptyString.
+    /// Attempts to create a new `NonEmptyString`.
     /// If the given `string` is empty, `Err` is returned, containing the original `String`, `Ok` otherwise.
-    pub fn new(string: String) -> Result<NonEmptyString, String> {
+    pub fn new(string: String) -> Result<Self, String> {
         if string.is_empty() {
             Err(string)
         } else {
             Ok(NonEmptyString(string))
         }
     }
+    /// Creates a new `NonEmptyString`, assuming it's not empty.
```

Comment:
In this wording, `it` seems to refer to the `NonEmptyString` that's created, but you're referring to the argument. I think you might have omitted part of the sentence you were going to type?